kratos/utilities/xml_utilities/xml_ostream_writer.cpp

```diff
@@ -0,0 +1,195 @@
+//    |  /           |
```

Comment:
Suggested C++ test: 

~~~c++
KRATOS_TEST_CASE_IN_SUITE(WriteElement, KratosCoreFastSuite)
{
    std::vector<std::pair<std::string, std::string>> attributes = {{"attribute1", "value1"}, {"attribute2", "value2"}};
    writer.WriteElement("TestElement", attributes, 1, false);
    KRATOS_CHECK_EQUAL(ss.str(), "   <TestElement attribute1=\"value1\" attribute2=\"value2\">\n");
}

KRATOS_TEST_CASE_IN_SUITE(CloseElement, KratosCoreFastSuite)
{
    writer.CloseElement("TestElement", 1);
    KRATOS_CHECK_EQUAL(ss.str(), "   </TestElement>\n");
}

KRATOS_TEST_CASE_IN_SUITE(WriteDataElementAscii, KratosCoreFastSuite)
{
    std::vector<std::pair<std::string, std::string>> attributes = {{"attribute1", "value1"}};
    std::vector<Expression::Pointer> expressions; // Assume you fill this with proper expressions
    writer.WriteDataElement("TestElement", attributes, expressions, 1);
    // Assuming that your expressions have the data, Check your result string here
}

KRATOS_TEST_CASE_IN_SUITE(WriteDataElementBinary, KratosCoreFastSuite)
{
    writer.SetOutputFormat(WriterFormat::BINARY);
    std::vector<std::pair<std::string, std::string>> attributes = {{"attribute1", "value1"}};
    std::vector<Expression::Pointer> expressions; // Assume you fill this with proper expressions
    writer.WriteDataElement("TestElement", attributes, expressions, 1);
    // Assuming that your expressions have the data, Check your result string here
}

KRATOS_TEST_CASE_IN_SUITE(WriteAttributes, KratosCoreFastSuite)
{
    std::vector<std::pair<std::string, std::string>> attributes = {{"attribute1", "value1"}};
    writer.WriteAttributes("TestElement", attributes, 1);
    KRATOS_CHECK_EQUAL(ss.str(), "   <TestElement attribute1=\"value1\"");
}
~~~

Reply:
done :)